repos/repo.go

```diff
@@ -0,0 +1,76 @@
+/* Copyright 2017 The Bazel Authors. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+   http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package repos
+
+import (
+	"fmt"
+	"path/filepath"
+	"sort"
+
+	"github.com/bazelbuild/bazel-gazelle/rules"
+	bf "github.com/bazelbuild/buildtools/build"
+)
+
+type repo struct {
+	name       string
+	importPath string
+	commit     string
+	remote     string
+}
+
+type byName []repo
+
+func (s byName) Len() int           { return len(s) }
+func (s byName) Less(i, j int) bool { return s[i].name < s[j].name }
+func (s byName) Swap(i, j int)      { s[i], s[j] = s[j], s[i] }
+
+var lockFileParsers = map[string]func(string) ([]repo, error){
+	"Gopkg.lock": importRepoRulesDep,
```

Comment:
I think you should add an extra step of indirection in here.
Have a function that peeks at a file to guess the "type" and returns a handler for that type.
The import them becomes a method on the returned object.
At the moment the only peeking is checking the base name, but you could also imagine looking inside the file for some magic to disambiguate. Also gives you a useful value to hang other things off in the future.

Reply:
Done